notebook/notebookapp.py

```diff
@@ -1100,6 +1109,13 @@ def start(self):
         
         This method takes no arguments so all configuration and initialization
         must be done prior to calling this method."""
+        try:
+            if os.geteuid() == 0 and not self.allow_root:
+                self.log.critical("Running as root is forbidden. Use --allow-root to bypass.")
+                self.exit(1)
+        except OSError as e:
+            pass
```

Comment:
Is there some condition that we're expecting to catch here? Why would `geteuid()` fail? If we're catching this, I think we should at least log the error so that there's an indication that something has gone wrong.